### PR TITLE
added CRUD functions

### LINKED ISSUE #31: All CRUD actions for courses

### PR DESCRIPTION
**Pull request by Blandon22**
## Changes Made
Updated Schema and added CRUD functions.

## Checklist

- [] Functionality works and does not implicate other code
- [] Written tests if applicable
- [] Error checking if applicable
- [] Code quality 

#### Issues Resolved

Closes #31

### CODE REVIEW

**Comment by Blandon22** [comment]
Keep iman's schema, its more flexible.

File: backend/controllers/courseController.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 const Course = require("../model/courseModel");
 const mongoose = require("mongoose");
 
+//Gets a specific course from mongoID
 const getCourse = async (req, res) => {
```
Comment (kunwar45) [inline backend/controllers/courseController.js:5]: change the function name to `showCourse`

File: backend/controllers/courseController.js [hunk 2]
```diff
@@ -17,11 +18,131 @@ const getCourse = async (req, res) => {
   });
 };
 
-const getCourses = async (req, res) => {
-  const { id } = req.params;
-  console.log(id);
+//Gets all Courses
+const getAllCourses = async (req, res) => {
```
Comment (kunwar45) [inline backend/controllers/courseController.js:22]: change the name to `indexCourses`

File: backend/routes/courses.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 const express = require("express");
 const router = express.Router();
-const { getCourse, getCourses } = require("../controllers/courseController");
+const { getCourse, getAllCourses,createCourse,deleteCourse,updateCourse } = require("../controllers/courseController");
```
Comment (kunwar45) [inline backend/routes/courses.js:3]: change these accordingly

File: backend/routes/courses.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 
 //get specified course
 router.get("/:id", getCourse);
```
Comment (kunwar45) [inline backend/routes/courses.js:6]: change accordingly

File: backend/routes/courses.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
-router.get("/", getCourses);
-
+router.get("/", getAllCourses);
```
Comment (kunwar45) [inline backend/routes/courses.js:7]: above^

**Review by kunwar45 — CHANGES_REQUESTED** [review]
Change the names of the functions to use standard REST notation of show and index

**Review by kunwar45 — APPROVED** [review]

**Pull request merged**